### PR TITLE
[db] Move db models to go-db module

### PR DESCRIPTION
**Pull request by laushinka**
## Description
Moves DB models to `components/gitpod-db/go`, so that they can be used by public-api but also usage.

Changes performed in this PR:
* Move all DB models from `components/usage/pkg/db into components/gitpod-db/go`
* Created a BUILD.yaml for `go`
* Update packages correspondingly
* Update imports correspondingly

## Related Issue(s)

New attempt at #14731
Requirement for #14602

## How to test

Build and unit tests

## Release Notes

```release-note
NONE
```

## Werft options:

- [ ] /werft with-local-preview
      If enabled this will build `install/preview`
- [ ] /werft with-preview
- [ ] /werft with-large-vm
- [ ] /werft with-integration-tests=all
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`

### CODE REVIEW

**Comment by roboquat** [comment]
@laushinka: Adding the "do-not-merge/release-note-label-needed" label because no release-note block was detected, please follow our [release note process](https://git.k8s.io/community/contributors/guide/release-notes.md) to remove it.

<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [kubernetes/test-infra](https://github.com/kubernetes/test-infra/issues/new?title=Prow%20issue:) repository.
</details>

**Comment by easyCZ** [comment]
@mads-hartmann We'll still need the leeway change to get this PR through. Do you have a ref for the fix so we can track, please?

File: components/gitpod-db/go/BUILD.yaml
Comment (andrew-farries) [inline components/gitpod-db/go/BUILD.yaml:22]: Is this what you mean by "Removes dependency from usage on gitpod-protocol for starting the DB" from the PR description?

File: components/gitpod-db/go/dbtest/conn.go
Comment (andrew-farries) [inline components/gitpod-db/go/dbtest/conn.go:31]:
```suggestion
	// These are static connection details for tests, started by `leeway build components/gitpod-db/go:init-testdb`.
```

**Review by andrew-farries — APPROVED** [review]
👍 Changes look mostly mechanical and I paired with @laushinka for some of this yesterday.

/hold for one small change.

File: components/gitpod-db/go/BUILD.yaml
Comment (laushinka) [reply components/gitpod-db/go/BUILD.yaml:22]: I copied the description over and realize I'm not sure what that last one meant with gitpod-protocol.. But this was moved from the `usage` package.

File: components/gitpod-db/go/dbtest/conn.go
Comment (laushinka) [reply components/gitpod-db/go/dbtest/conn.go:31]: Thanks!

File: components/gitpod-db/go/BUILD.yaml
Comment (andrew-farries) [reply components/gitpod-db/go/BUILD.yaml:22]: Not sure what it means, actually 😕 . I don't see any dependencies on `gitpod-protocol` being removed from `usage` - only the `dbtest-init` package moving from `usage` to `gitpod-db/go`.

Maybe @easyCZ can explain?

Comment (easyCZ) [reply components/gitpod-db/go/BUILD.yaml:22]: This made sense in the context of the previous PR, no longer makes sense. What it does is this:
* Moves the db definitions to `gitpod-db/go`
* Usage & Public API now depend on `gitpod-db/go`

Comment (easyCZ) [reply components/gitpod-db/go/BUILD.yaml:22]: I've updated the description

**Comment by mads-hartmann** [comment]
@laushinka @easyCZ It looks like the build is fine without the Leeway change now? ☺️ I opened a  [PR](https://github.com/gitpod-io/leeway/pull/121) to fix it anyway as it was indeed a bug in Leeway - @andrew-farries have a look at the PR description if you want to learn more about ephemeral packages.

**Comment by andrew-farries** [comment]
Thanks for the [Notion doc](https://www.notion.so/gitpod/Leeway-and-ephemeral-packages-d06d0ab9cdd84f3a9e4587033c0fb2b4#477d55962ddb406da5bf0e7533a26006) @mads-hartmann.

If ephemeral packages are something that we intend to keep using, perhaps your PR (or a followup) could add documentation about them to leeway's `README.md`?

**Comment by mads-hartmann** [comment]
> If ephemeral packages are something that we intend to keep using, perhaps your PR (or a followup) could add documentation about them to leeway's README.md?

I will try to get clarity around the future of ephemeral packages with the team and Chris. If we keep them around I will for sure update the documentation 🧡

**Comment by easyCZ** [comment]
@mads-hartmann I guess it's not really needed. Which raises the Q why was it needed before? The changes here are the same as in the original PR.

**Comment by mads-hartmann** [comment]
@easyCZ I think you're seeing the problem I'm describing in [the PR](https://github.com/gitpod-io/leeway/pull/121). Looking at the logs of one of the failed jobs ([job 12](https://werft.gitpod-dev.com/job/gitpod-build-mp-common-move-db-models.12/raw)) you can see that it didn't need to rebuild `components/common-go:lib`(`cached remotely (downloaded)  components/common-go:lib`) which means it wouldn't "build" the ephemeral package `init-testdb`. Looking at the successful build in this branch ([job 24](https://werft.gitpod-dev.com/job/gitpod-build-lau-move-db-models-godb.24/raw)) you can see it does need to build `components/gitpod-db/go:lib` so the ephemeral package `components/gitpod-db:dbtest-init` will be built too, which means that any package that depends on `components/gitpod-db/go:lib` will built successfully.

So I think without [the PR](https://github.com/gitpod-io/leeway/pull/121) we'll see flaky builds depending on whether or not Leeway needs to build  `components/gitpod-db/go:lib` or not. I'm not 100% sure on this, but it's my best guess ☺️

**Comment by easyCZ** [comment]
@mads-hartmann Looks like you've managed to land https://github.com/gitpod-io/leeway/pull/121, what changes do we need to make to this PR to use it?

**Comment by mads-hartmann** [comment]
@easyCZ I'll open a PR to bump the Leeway version *everywhere* now. I'll ping you as soon as it's ready, and once that's in we can merge this PR ☺️

**Comment by mads-hartmann** [comment]
@easyCZ PR [here](https://github.com/gitpod-io/gitpod/pull/14812) but the main build is broken since a few hours ago, so I'm going down that rabbit hole first ☺️

**Comment by mads-hartmann** [comment]
@easyCZ It's in main :shipit:

**Review by MrSimonEmms — APPROVED** [review]

**Review by Furisto — APPROVED** [review]

**Pull request merged**
